File: roman_to_integer.py
class Solution:
    def romanToInt(self, s: str) -> int:
        romans = {
            "I": 1,
            "V": 5,
            "X": 10,
            "L": 50,
            "C": 100,
            "D": 500,
            "M": 1000,
        }
        sum = 0
        n = len(s)
        i = 0
        while i < n:
            if i < n - 1 and romans[s[i]] < romans[s[i + 1]]:
                sum += romans[s[i + 1]] - romans[s[i]]
                i += 2
            else:
                sum += romans[s[i]]
                i += 1
        return sum


def main():
    ch = input("enter the roman")
    print(Solution().romanToInt(ch))


if __name__ == "__main__":
    main()

File: roman_to_integer.cpp
#include <iostream>
#include <string>
#include <unordered_map>

class Solution {
public:
  int romanToInt(std::string s) {
    std::unordered_map<std::string, int> romans = {
        {"I", 1},   {"V", 5},   {"X", 10},  {"L", 50},
        {"C", 100}, {"D", 500}, {"M", 1000}};
    int sum = 0;
    int n = s.size();
    int i = 0;
    while (i < n) {
      std::string current(1, s[i]);
      std::string next = (i < n - 1) ? std::string(1, s[i + 1]) : "";

      if (i < n - 1 && romans[current] < romans[next]) {
        sum += romans[next] - romans[current];
        i += 2;
      } else {
        sum += romans[current];
        i++;
      }
    }
    return sum;
  }
};

int main(int argc, char *argv[]) {

  Solution test;
  std::string roman;
  std::cin >> roman;
  std::cout << test.romanToInt(roman);

  return 0;
}
